File: freecodecamp/daily-code-challenge/2026-01-20.py
def to_consonant_case(s):
    vowels = {'a', 'e', 'i', 'o', 'u'}

    return ''.join(
        '_' if ch == '-' else
        ch.lower() if ch.lower() in vowels else
        ch.upper() if ch.isalpha() else
        ch
        for ch in s
    )

File: freecodecamp/daily-code-challenge/2026-01-20.js
function toConsonantCase(str) {
  const vowels = ['a', 'e', 'i', 'o', 'u'];

  return str
    .split('')
    .map(char => {
      if (char === '-') {
        return '_';
      }

      if (vowels.includes(char.toLowerCase())) {
        return char.toLowerCase();
      }

      if (/[a-z]/i.test(char)) {
        return char.toUpperCase();
      }

      return char;
    })
    .join('');
}
